tests/conftest.py

```diff
@@ -31,3 +34,26 @@ def env_save():
     ]
     set_xor = set(environment_pre).symmetric_difference(set(environment_post))
     assert len(set_xor) == 0, f"Detected differences in environment: {set_xor}"
+
+
+# Enable to find a test that is changing the source files or chdir-ing without going back
+@pytest.fixture(autouse=False)
+def detect_file_chages(source_root):
+    # Uses git status to check that there were no changes to the source tree while running tests
```

Comment:
Eh, I guess valid concern, but could we maybe have a CI and test setup where this has a low to no probability of  happening instead?

Reply:
That would be the goal. Hopefully that is fixed now. I'll just remove this code.